### ISSUE TITLE
Use outside of command line

### ISSUE THREAD

**Issue by joshribakoff** [cakephp/phinx#137]
Basically I just need to make a file "test.php" and when I load that in my browser, it needs to run my migrations.

Phinx is great for all my work projects, and personal projects where I don't distribute the code, but now I want to use it in a project I distribute. We don't use symfony, and 99% of my users don't have shell access, and calls to exec() aren't guaranteed to work.

So I start hacking it in ways its not intended to work:

``` php
<?php
ini_set('display_errors',1);
error_reporting(E_ALL);
$_SERVER['argv'] = array(
    0=>'migrate',
);
require_once 'app/code/local/Elite/vendor/vehiclefits/library/vendor/autoload.php';
$app = new Phinx\Console\PhinxApplication('0.2.8');
var_dump($app->run());
```

This gives me no output. Diving into the internals it looks like Symfony\Component\Console\Application has an exit statement on line 149. Personally this is why I prefer Zend, Symfony advocates praise it because "it works out of the box" with less configuring, which is true -- but now I need to "make it work out of the box" and it doesn't work at all.

Hacking around even more it looks like you can pass in input/output objects, and there's a flag called "auto exit" (what a pain in the neck):

``` php
<?php
ini_set('display_errors',1);
error_reporting(E_ALL);


require_once 'app/code/local/Elite/vendor/vehiclefits/library/vendor/autoload.php';

$input = new \Symfony\Component\Console\Input\StringInput('help');

$output = new \Symfony\Component\Console\Output\ConsoleOutput();

$app = new Phinx\Console\PhinxApplication('0.2.8');
$app->setAutoExit(false);
var_dump($app->run($input, $output));
```

Now its just exiting with code 1. Looks like maybe its writing to stderr which effectively goes nowhere when developing in the browser.

Personally this is why I don't like symfony. Its great for rapid app development, but this is an example of where its lacking in extensibility/re-use. The Zend console component does nothing more than parse the command line arguments for you, leaving everything else up to the developer. The symfony console component imposes this concept of an app/kernel, input & output wrappers that pipe stuff to no mans land, etc..

In summary, phinx is great, but I'm not a symphony fan in case you didn't notice. I would like to see phinx decoupled from this invasive symfony framework so I can just instantiate a class & use it to migrate the database, without hacking around with $_SERVER and trying to force it to send its output to stdout.
